### PR TITLE
kernel: refact

### PR DESCRIPTION
**Pull request by Ylarod**

### CODE REVIEW

File: kernel/allowlist.c [hunk 1]
```diff
@@ -256,39 +240,37 @@ void ksu_prune_allowlist(bool (*is_uid_exist)(uid_t, void *), void *data)
...
 }
 
-bool ksu_allowlist_exit(void)
+void ksu_allowlist_exit(void)
 {
-	return true;
+	struct perm_data *np = NULL;
+	struct perm_data *n = NULL;
+
+	persistent_allow_list();
```
Comment (tiann) [inline kernel/allowlist.c:267]: call `do_persistent_allow_list` directly?

File: kernel/core.c [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+#endif
+}
+
+void ksu_core_exit(void)
+{
+#ifndef MODULE
+	pr_info("ksu_kprobe_exit\n");
+	ksu_kprobe_exit();
+#endif
+}
```
Comment (tiann) [inline kernel/core.c:354]: rename file to `core_hook.c`? `core` is a bit broad.

File: kernel/ksu.c [hunk 3]
```diff
@@ -328,6 +41,7 @@ int __init kernelsu_init(void)
 
 #ifdef CONFIG_KPROBES
 	enable_sucompat();
```
Comment (tiann) [inline kernel/ksu.c:43]: prefix it with `ksu_`?

File: kernel/ksud.c [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+#else
+static bool vfs_read_hook = true;
+static bool execveat_hook = true;
+#endif
+
+int ksu_handle_execveat_ksud(int *fd, struct filename **filename_ptr,
+			     void *argv, void *envp, int *flags)
+{
+#ifndef CONFIG_KPROBES
+	if (!execveat_hook) {
```
Comment (tiann) [inline kernel/ksud.c:54]: what is this for?

File: kernel/core.c [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+	if (arg2 == CMD_BECOME_MANAGER) {
+		// quick check
+		if (is_manager()) {
+			if (copy_to_user(result, &reply_ok, sizeof(reply_ok))) {
+				pr_err("become_manager: prctl reply error\n");
+			}
+			return 0;
+		}
+		if (ksu_is_manager_uid_valid()) {
+			pr_info("manager already exist: %d\n", ksu_manager_uid);
```
Comment (tiann) [inline kernel/core.c:133]: get_ ksu_manager_uid()?

File: kernel/ksud.c (hunk 4 above)
Comment (Ylarod) [reply kernel/ksud.c:54]: like unregister kprobe, this is for non-gki

**Review by tiann — CHANGES_REQUESTED** [review]

**Review by tiann — APPROVED** [review]

**Comment by tiann** [comment]
Maybe we should also bump ksu version?

**Comment by Ylarod** [comment]
bump after merge?

**Comment by tiann** [comment]
> bump after merge?

ok

**Pull request merged**